src/pgw_s5s8_proxy.erl

```diff
@@ -621,7 +645,26 @@ handle_response(#proxy_request{request = ReqKey} = _ReqInfo,
     ?LOG(warning, "Unknown Proxy Response ~p", [Response]),
 
     gtp_context:request_finished(ReqKey),
-    keep_state_and_data.
+    keep_state_and_data;
+
+handle_response(_, _, #gtp{type = delete_bearer_request}, shutdown, _) ->
```

Comment:
Looks like can be from line 650 till 667 change to:
```erlang
handle_response(_, _, #gtp{type = Type}, shutdown, _)
  when Type =:= delete_bearer_request; Type =:= delete_session_request ->
    keep_state_and_data;

handle_response({Direction, From}, timeout, #gtp{type = Type}, shutdown_initiated, Data) 
  when Type =:= delete_bearer_request; Type =:= delete_session_request ->
    session_teardown_response({error, timeout}, Direction, From, Data);

handle_response({Direction, From}, #gtp{type = Type}, _, shutdown_initiated, Data) 
  when Type =:= delete_bearer_response; Type =:= delete_session_response ->
    session_teardown_response(ok, Direction, From, Data).
```

Reply:
Intention was to keep this a identical as possible to the similar code in ggsn_gn_proxy. But using the shorter version doesn't hurt.